pkg/graphics/camera_2d.go

```diff
@@ -66,6 +67,19 @@ func (c *Camera2D) SetFlipVertical(flip bool) {
 	c.matrixDirty = true
 }
 
+// SetVisibleArea configures the camera to make the specified area completely visible, position and zoom are changed accordingly
+func (c *Camera2D) SetVisibleArea(x1 float32, y1 float32, x2 float32, y2 float32) {
+	width := float32(math.Abs(float64(x2 - x1)))
+	height := float32(math.Abs(float64(y2 - y1)))
+	zoom := float32(math.Min(float64(c.width/width), float64(c.height/height)))
+	c.SetZoom(zoom)
+	if c.centered {
+		c.SetPosition(x1+width/2, y1+height/2)
```

Comment:
we should either assert that `x1<x2` or `SetPosition` based on `x2` if it's smaller